src/OgAccess.php

```diff
@@ -112,18 +111,25 @@ public static function userAccess(EntityInterface $group, $operation, AccountInt
     // To reduce the number of SQL queries, we cache the user's permissions
     // in a static variable.
     if (!$pre_alter_cache) {
-      $permissions = array();
+      $permissions = [];
+      $user_is_group_admin = FALSE;
 
       foreach (Og::getUserMemberships($user) as $membership) {
-
         foreach ($membership->getRoles() as $role) {
 
           /** @var $role RoleInterface */
-          $permissions = array_unique(array_merge($permissions, $role->getPermissions()));
+          $permissions = array_merge($permissions, $role->getPermissions());
+
+          // Set an is_admin flag.
+          if ($role->isAdmin()) {
+            $user_is_group_admin = TRUE;
+          }
         }
       }
 
-      static::setPermissionCache($group, $user, TRUE, $permissions, $cacheable_metadata);
+      $permissions = array_unique($permissions);
```

Comment:
@amitaibu a single `array_unique` is kept here. Then we are just doing it once, instead of once per role.